constructorlab.js

```diff
@@ -0,0 +1,260 @@
+// # Constructor Exercises
+
+// ## Question 1
+
+// a. Write a constructor function called `Person` that has 3 properties: a first name, a last name and a middle name. Create 2 instances of a `Person`. Print one of their first names.
+function Person(firstName, middleName, lastName) {
+    this.firstName = firstName;
+    this.middleName = middleName;
+    this.lastName = lastName;
+}
+let person1 = new Person("Ashya", "Lamina", "Manning")
+let person2 = new Person("David", "Peter", "Maravillas")
+let person3 = new Person("Jon", "Jay", "Jones")
+console.log(person1.firstName)
+
+// b. Write a prototype function in `Person` called `fullName` that will return a formatted string of an instance's full name. Call this method on both the instances you created in part a.
+Person.prototype.fullName = function() {
+    console.log(this.firstName + " " + this.middleName + " " + this.lastName)
+}
+person1.fullName()
+
+
+
+// ## Question 2
+
+// a. Create a constructor function called `Book` that has properties `title`, `author` and `rating`. Create some instances of `Book`.
+function Book(title, author, rating) {
+    this.title = title;
+    this.author = author;
+    this.rating = rating;
+}
+let book1 = new Book("Harry Potter", "J.K. Rowling", 8)
+let book2 = new Book("Radical Candor", "Kim Scott", 10)
+let book3 = new Book("The Great Gasby", "F. Scott Fitzgerald", 6)
+// b. Add a prototype function to `Book` called `isGood` that returns `true` if its rating is greater than or equal to 7
+Book.prototype.isGood = function() {
+    if (this.rating >= 7) {
+        return true
+    } else {
+        return false
+    }
+}
+console.log(book1.isGood())
+
+
+
+// ## Question 3
+
+// a. Create a `Dog` constructor function with four properties: `name (string), breed (string), mood (string), and hungry (boolean)`.
+function Dog(name, breed, mood, hungry) {
+    this.name = name;
+    this.breed = breed;
+    this.mood = mood;
+    this.hungry = hungry;
+}
+let happy = new Dog("Happy", "pug", "happy", true)
+let chip = new Dog("Chip", "dalmation", "hyper", false)
+
+// b. Add a prototype function called `playFetch`. It should set the dog's `hungry` property to `true`, set its mood property to `playful`, and log "Ruff!"
+Dog.prototype.playFetch = function() {
+    this.hungry = true;
+    this.mood = "playful";
+    console.log("Ruff")
+}
+happy.playFetch()
+console.log(happy)
+
+// c. Add a prototype function called `feed`. If the dog is hungry, it should set `hungry` to `false` and print "Woof!" If the dog is not hungry, it should log "The dog doesn't look hungry"
+Dog.prototype.feed = function() {
+    if (this.hungry === true) {
```

Comment:
Because this.hungry evaluates to a boolean there is no need to do the === true. It's not wrong, just not needed.